src/main/java/com/yanna/stepanova/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,75 @@
+package com.yanna.stepanova.controller;
+
+import com.yanna.stepanova.dto.cartitem.CartItemDto;
+import com.yanna.stepanova.dto.cartitem.CartItemQuantityRequestDto;
+import com.yanna.stepanova.dto.cartitem.CreateCartItemRequestDto;
+import com.yanna.stepanova.dto.shoppingcart.ShoppingCartDto;
+import com.yanna.stepanova.model.User;
+import com.yanna.stepanova.service.CartItemService;
+import com.yanna.stepanova.service.ShoppingCartService;
+import com.yanna.stepanova.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping Cart manager", description = "Endpoints for managing user's shopping cart")
+@RequiredArgsConstructor
```

Comment:
```suggestion
@RequiredArgsConstructor
@Validated
```
because your are using `@Positive` annotation for Long id